esx_service/vmdk_ops.py

```diff
@@ -402,8 +402,15 @@ def vol_info(vol_meta, vol_size_info, datastore):
 # Return error, or None for OK
 def removeVMDK(vmdk_path):
     logging.info("*** removeVMDK: %s", vmdk_path)
-    cmd = "{0} {1}".format(VMDK_DELETE_CMD, vmdk_path)
 
+    # Check the current volume status
+    attached, uuid, attach_as = getStatusAttached(vmdk_path)
+    if attached:
```

Comment:
@govint - good point about  `handle_stale_attach()` reuse, it seems to be simple way to improve the check. However - I am very concerned that it's getting too complex with too many if-then-else and race potentials. E.g. what would happen if  the disk is  attached to a VM registered on a different ESX host ? Or if VM with the disk is powered off, but for some reason KV does not show attached (I know, it "should not" happen, but the experience says that bugs exist and it could happen).